### PR TITLE
Reduce actor submission python overhead

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

These simple changes bring down actor submission from 54us to 19us (only the python side).

Ported from https://github.com/ray-project/ray/pull/4522

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/_raylet.pyx [hunk 1]
```diff
@@ -689,20 +689,20 @@ cdef class CoreWorker:
...
-            prepare_args(args, &args_vector)
-
-            with nogil:
-                check_status(self.core_worker.get().SubmitActorTask(
-                      c_actor_id,
-                      ray_function,
-                      args_vector, task_options, &return_ids))
-
-            return VectorToObjectIDs(return_ids)
+        # with profiling.profile("submit_task"):
```
Comment (simon-mo) [inline python/ray/_raylet.pyx]: is removing profiling here ok? Is the profiling part a latency bottleneck?

Comment (pcmoritz) [reply python/ray/_raylet.pyx]: yeah this was just for testing, I'm bringing it back :)

Turns out if we replace the event loop call with a lock, it is fast.

File: src/ray/core_worker/profiling.h [hunk 2]
```diff
@@ -1,9 +1,11 @@
 #ifndef RAY_CORE_WORKER_PROFILING_H
 #define RAY_CORE_WORKER_PROFILING_H
 
+#include "absl/synchronization/mutex.h"
```
Comment (simon-mo) [inline src/ray/core_worker/profiling.h]: Does this need to be added to bazel?

**Review by simon-mo — APPROVED** [review]

**Comment by pcmoritz** [comment]
Latest numbers with this PR (`core_worker.get().SubmitActorTask` commented out):

After:

```
In [3]: @ray.remote 
   ...: class C: 
   ...:     def f(self): 
   ...:         return 42 
   ...:                                                                                                                                                                              

In [4]: c = C.remote()                                                                                                                                                               

In [5]: %timeit c.f.remote()                                                                                                                                                         
20.5 µs ± 1.29 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Before:

```
In [8]: %timeit c.f.remote()
69.6 µs ± 5.05 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

There are some more optimizations from #4522 that can be ported now that the core worker execution interface and the kwargs PR are merged (before there would have been a bunch of conflicts). Will do that in a followup PR.

**Comment by pcmoritz** [comment]
Jenkins failure was introduced by https://github.com/ray-project/ray/pull/5783

**Pull request merged**